packages/core-components/src/layout/SignInPage/guestProvider.tsx

```diff
@@ -20,39 +20,108 @@ import Button from '@material-ui/core/Button';
 import { InfoCard } from '../InfoCard/InfoCard';
 import { GridItem } from './styles';
 import { ProviderComponent, ProviderLoader, SignInProvider } from './types';
+import { ProxiedSignInIdentity } from '../ProxiedSignInPage/ProxiedSignInIdentity';
+import { discoveryApiRef, useApi } from '@backstage/core-plugin-api';
 import { GuestUserIdentity } from './GuestUserIdentity';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { ResponseError } from '@backstage/errors';
 
-const Component: ProviderComponent = ({ onSignInStarted, onSignInSuccess }) => (
-  <GridItem>
-    <InfoCard
-      title="Guest"
-      variant="fullHeight"
-      actions={
-        <Button
-          color="primary"
-          variant="outlined"
-          onClick={() => {
-            onSignInStarted();
-            onSignInSuccess(new GuestUserIdentity());
-          }}
-        >
-          Enter
-        </Button>
+const getIdentity = async (identity: ProxiedSignInIdentity) => {
+  try {
+    const identityResponse = await identity.getBackstageIdentity();
+    return identityResponse;
+  } catch (error) {
+    if (
+      error instanceof ResponseError &&
+      error.cause.name === 'NotFoundError'
```

Comment:
```suggestion
      error.name === 'ResponseError' &&
      (error as ResponseError).cause.name === 'NotFoundError'
```

`instanceof` is dangerous, since you don't know that the `identity` implementation uses the same version of `@backstage/errors` as this package. This check is kinda the best we can do for now, but a more thorough `ResponseError.is(...)` or something like that might be nice